stl/inc/shared_mutex

```diff
@@ -313,43 +442,68 @@ public:
         return _Owns;
     }
 
+    _Pre_satisfies_(_Owns == true)
+    _Pre_satisfies_(_Pmtx != nullptr)
+    _Post_satisfies_(_Owns == false)
+    _When_(_Owns == false, _Analysis_noreturn_)
+    _When_(_Pmtx == nullptr, _Analysis_noreturn_)
+    _When_(_Owns == true, _Releases_shared_lock_(*_Pmtx))
+    _When_(_Owns == true, _Releases_nonreentrant_lock_(*_Pmtx))

```

Comment:
I believe that these should be unconditional. In every other case, after stating the preconditions, and saying `_When_(PRECONDITION VIOLATION, _Analysis_noreturn_)`, the remaining annotations assume a valid call. For example, see `shared_lock::lock()` above.